lightning/src/ln/channelmanager.rs

```diff
@@ -3156,6 +3198,17 @@ impl<M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelManager<M, T, K, F
 				}
 			}
 		}
+		{
+			let mut retryable_htlcs = Vec::new();
+			mem::swap(&mut retryable_htlcs, &mut self.pending_outbound_payments.retryable_htlcs.lock().unwrap());
+			for (payment_id, params) in retryable_htlcs {
+				// TODO: do we need to generate a PendingHTLCsForwardable event?
+				if let Err(e) = self.retry_payment_auto_route(payment_id, params) {
+					log_trace!(self.logger, "Errored retrying payment, marking as abandoned: {:?}", e);
+					self.abandon_payment(payment_id);
```

Comment:
@TheBlueMatt Are we moving to requiring users to call `abandon_payment`? You referenced a race condition with this method offline, could you elaborate?